dashboard/app/controllers/registrations_controller.rb

```diff
@@ -49,6 +61,22 @@ def create
       storage_id = take_storage_id_ownership_from_cookie(current_user.id)
       current_user.generate_progress_from_storage_id(storage_id) if storage_id
     end
+
+    sign_up_type = session[:sign_up_type]
+    sign_up_type ||= resource.email ? 'email' : 'other'
+    if session[:sign_up_tracking_expiration]&.future?
+      result = resource.persisted? ? 'success' : 'error'
+      tracking_data = {
+        study: 'account-sign-up',
+        event: "#{sign_up_type}-sign-up-#{result}",
+        data_string: session[:sign_up_uid],
+        data_json: {
+          detail: resource.to_json,
+          errors: resource.errors&.messages
```

Comment:
should this be `full_messages` instead of `mesages`? not sure that it matters, just changes the formatting of the error messages